src/os/bluestore/BlueStore.cc

```diff
@@ -1391,6 +1391,10 @@ void BlueStore::BufferSpace::finish_write(Cache* cache, uint64_t seq)
 {
   std::lock_guard<std::recursive_mutex> l(cache->lock);
 
+  if (writing.empty() || writing.rbegin()->seq < seq) {
```

Comment:
yes, that's what I mean. Original logic assumes 'writing' list is sorted but that's not 100% true.